### ISSUE TITLE
Gitlab/Github/Bitbucket Source detection

### ISSUE THREAD

**Issue by brendo** [squidfunk/mkdocs-material#57]
## Description

Using the `rework` branch, the `repo_url` is used to determine the platform. However this detection does not work for platforms if they are self hosted, such as Gitlab.
### Expected behavior

The developer is able to explicitly state the platform through some means.
### Actual behavior

The platform is determined from looking at the URL.
### Steps to reproduce the bug
1. Set repo url to something that does not include `gitlab`, `github` or `bitbucket`
2. Note that there is no special treatment to the resulting page.

Perhaps this could be set via the `extras` configuration as `repo_type`?

(note that there's also a bug where if the `repo_url` is not specified at all `mkdocs build` will fail with `TypeError: argument of type 'NoneType' is not iterable`)
### Package versions
- Python: 2.7.12
- MkDocs: mkdocs, version 0.15.3
- Material: `rework` branch @ 7b585f191ac8a168eb4bcabe7c8cd74fa805e33d

**Comment by squidfunk** [comment]
Sounds good to me! I will prepare something for 1.0.x.

**Comment by squidfunk** [comment]
> (note that there's also a bug where if the repo_url is not specified at all mkdocs build will fail with TypeError: argument of type 'NoneType' is not iterable)

Fixed in eaeba59 (rework branch)

**Comment by squidfunk** [comment]
Added `repo_type` option to `extra` namespace in the rework branch:

``` yaml
extra.repo_type = github
extra.repo_type = gitlab
extra.repo_type = bitbucket
```

Will be released with 1.0.0.

**Comment by brendo** [comment]
Great! I can confirm `extra.repo_icon` variable works, and the `NoneType` error has also been resolved.
